File: boruvka/Go/boruvka.go
package main

import "fmt"

// representacion simplificada de 4 Grafos: Vertice_Origen, Ponderacion, Vertice_Destino
var grafos [4][3]uint8

func main() {

	fmt.Println("\nCarga de 4 grafos\n")

	for fila := 0; fila < len(grafos); fila++ {
		fmt.Println("\tIngrese origen, ponderacion, destino: ")
		for columna := 0; columna < len(grafos[fila]); columna++ {
			fmt.Print("\t\t")
			fmt.Scanln(&grafos[fila][columna])
		}
	}

	var menor uint8 = grafos[0][1]

	var indice uint32

	for grafo_seleccionado := 0; grafo_seleccionado < len(grafos); grafo_seleccionado++ {
		if grafo_seleccionado == len(grafos)-1 {
			break
		} else if menor >= grafos[grafo_seleccionado+1][1] {
			// Compara ponderacion, la guarda junto a la fila donde está el grafo
			menor = grafos[grafo_seleccionado+1][1]
			indice = uint32(grafo_seleccionado)
		}
	}

	fmt.Println("\nEl grafo de menor ponderacion es ", grafos[indice][0], ", ", menor, ", ", grafos[indice][2])
}

File: boruvka/Python/boruvka.py
#!/usr/bin/python3

from grafon import grafo

ponderados = []

print('Implementación simple de algoritmo de bor\u016fvka\n')

for ingresado in range(1, 5):
    ponderados.append(grafo(input('\n\tIngrese vertice origen: '), input('\n\tIngrese ponderación: '), input('\n\tIngrese vertice destino: ')))
    print(f'\n\t\t[ Grafo {ingresado.__str__()} ingresado ]\n')

menor = ponderados[0]

for comparar in ponderados:
    if menor.ponderado >= comparar.ponderado:
        menor = comparar

print(f'\n\nEl grafo(origen, ponderado, destino) con menor ponderación es ({menor.vertice_origen.__str__()}, {menor.ponderado.__str__()}, {menor.vertice_destino.__str__()})\n')